add navigation controller

BannikovRuslan opened the pull request.

BannikovRuslan committed:
Merge remote-tracking branch 'origin/develop' into feature-232254-add-navigation-controller

BannikovRuslan committed:
feat: add navigation controller

fix: code refactoring

BannikovRuslan committed:
fix: code smells

AntonIV8 commented on src/AndroidODataOfflineSample/app/build.gradle:
```diff
@@ -92,4 +92,12 @@ dependencies {
     testImplementation("androidx.room:room-testing:$room_version") // Test helpers
+
+    // Navigation
+    def nav_version = "2.5.3"
```
тут наверное не надо выносить в переменную. У нас только одна зависимость ее использует, а не неск.
def hilt_compiler_version также

AntonIV8 commented on src/AndroidODataOfflineSample/app/src/main/java/com/flexberry/androidodataofflinesample/navigation/AppNavigator.kt:
```diff
@@ -0,0 +1,45 @@
+    )
+
+    fun tryNavigateBack(
+        route: String? = null,
```
А вот эти try функции, это в оф источниках так рекомендуют ?

BannikovRuslan replied:
Это реализуются 4 функции, связанные с navigationChannel.
Выполнено на базе [этого примера](https://medium.com/nuances-of-programming/%D0%BA%D0%B0%D0%BA-%D0%BE%D0%BF%D1%82%D0%B8%D0%BC%D0%B8%D0%B7%D0%B8%D1%80%D0%BE%D0%B2%D0%B0%D1%82%D1%8C-%D0%BD%D0%B0%D0%B2%D0%B8%D0%B3%D0%B0%D1%86%D0%B8%D1%8E-%D0%B2-jetpack-compose-2bea946d51e4)

AntonIV8 commented on src/AndroidODataOfflineSample/app/src/main/java/com/flexberry/androidodataofflinesample/navigation/AppNavogatorImplement.kt:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    override fun tryNavigateBack(route: String?, inclusive: Boolean) {
+        navigationChannel.trySend(
```
Опять про try. Вроде ведь ничем не отличаются даже функции

AntonIV8 commented on src/AndroidODataOfflineSample/app/src/main/java/com/flexberry/androidodataofflinesample/navigation/BaseNavigation.kt:
```diff
@@ -0,0 +1,85 @@
+        // A surface container using the 'background' color from the theme
+        Surface(
+            modifier = Modifier.fillMaxSize(),
+            color = MaterialTheme.colorScheme.background
```
А вот это точно тут должно быть ? странно просто, что визуал мы теперь будем в BaseNavigation настраивать.

Относится ко всему Surface

BannikovRuslan replied:
Уберу в MainActivity

BannikovRuslan committed:
Merge remote-tracking branch 'origin/develop' into feature-232254-add-navigation-controller

BannikovRuslan committed:
fix: некоторые правки с учётом предыдущего слияния с develop

BannikovRuslan committed:
fix: filename

BannikovRuslan committed:
fix: build.gradle, удалены переменные, связанные с номерами версий + code smells.
